src/libstd/thread/mod.rs

```diff
@@ -208,6 +208,14 @@ pub use self::local::{LocalKey, AccessError};
 #[unstable(feature = "libstd_thread_internals", issue = "0")]
 #[doc(hidden)] pub use self::local::os::Key as __OsLocalKeyInner;
 
+/// Function used for resetting the main stack guard address after setrlimit().
+/// This is POSIX specific and unlikely to be directly stabilized.
+#[unstable(feature = "rustc_stack_internals", issue = "0")]
+pub unsafe fn update_stack_guard() {
+    let main_guard = imp::guard::init();
```

Comment:
Hm are we sure this works? On Unix platforms that aren't Linux it looks like we use `mmap` to make the last page of the stack non-mapped so we fault, but it looks like this doesn't reset that? I think we'll want to undo that call to `mmap`, right?

Also mind moving this to a perma-unstable location like `std::rt`?

Reply:
Yeah it seems you're right, we need to revert the changes of mmap.

Though, std::rt seems to be a place for startup related functions only, so I'm not sure if it's a good place of placing this function.

Also, do you know if an unstable book doc file is needed for this kind of features?

Reply:
Oh it's ok `std::rt` is more of a dumping ground than a principled location! And yeah it's fine to skip out on the unstable book for something like this